azure/components/lambda/setup.ftl

```diff
@@ -123,12 +123,12 @@
                         productName,
                         occurrence,
                         function.Name + ".zip",
-                        getReference("ResourceGroup")
+                        getReference(function.Id, function.Name, "ResourceGroup")
```

Comment:
A general question about the getReference usage. You added the handling for param1 of getReference to be a structure with both Id and Name (eg  getReference({"Id": function.Id, "Name": function.Name}, "ResourceGroup") refer L138 in resource.ftl), but also added Name as an optional param2. I don't understand why you'd do both. What have I missed?

Reply:
Passing an object in the way you've mentioned was just to enable a shorthand - It's really only of use in the setup or state routines and is a reflection of how I've been writing lately. 

I find it clearer to assign a variable to a resource and then access the `Id` or `Name` on that resource as I need them, rather than assign one variable for each as has been done in the past. It's also more useful as I've begun assigning the `resource.Reference` values in the state routine, so if something is referenced many times (common) then there is only one call to `getReference` instead of many. If I only had `resourceId` and `resourceName` variables, accessing this reference would be a bit more annoying. 

```
[#local bucket = resources["bucket"] ]
[#local exampleString = "example" + bucket.Id + bucket.Name]
```

```
[#local bucketId = resources["bucket"].Id ]
[#local bucketName = resources["bucket"].Name ]
[#local exampleString = "example" + bucketId + bucketName]
```

So typically,
* In state/setup routines, I want to be able to pass either an Id or an entire resource definition to `getReference`
* If it's not in those routines, I won't have access to these objects necessarily but want to do the same work